README.rst

```diff
@@ -308,18 +467,6 @@ and for html:
 
 This is more boilerplate though.
 
-You can also refactor ``parse`` to have a single argument:
-
-.. code-block:: python
-
-    @dataclass
-    class Response:
-        url: str
-        html: str
-
-    def parse(response: Response):
-        # ...
-
```

Comment:
Why do you think this approach is problematic?

Reply:
Well, it is not problematic, but as an example is confusing. The way it is isn't solving the problem of having two str arguments unless the `Response` is considered an externally provided class. Otherwise, the right code would have been:

```py
    class HTML(str):
        pass

    class URL(str):
        pass

    @dataclass
    class Response:
        url: HTML
        html: URL

    def parse(response: Response):
        # ...
```

Therefore boilerplate is not reduced.

By the other hand I think is already clear in the documentation that you can pack dependencies into classes (e.g. car packs both engine + wheels. 

In any case, I'm going to introduce another section to explain how to use `andi` to build callback based framework and I think there I can introduce the `Response` concept as externally provided.

Reply:
Yes, I was assuming in this example Response is externally provided, like HTML or URL - you may decide to provide it all at once.

Reply:
I have added the section `Examples: callback based frameworks` where I think this is more or less covered. Let me know if something should be clarified further or new content added.